Support FDP v1 (mongodb) and v2 (postgresql)

dennisvang opened the pull request:
The [FDP `develop` branch][1] (upcoming v2) uses postgres, whereas current [releases v1.16 and v1.17][2] use mongodb.

This PR adapts the docker compose configuration to support both mongodb and postgres.

The Docker compose template has been replaced by explicit compose files for fdp v1 and v2 (develop).

The Makefile automatically selects the correct compose file based on the value of `SERVER_VERSION`.

If `SERVER_VERSION` is not set, it defaults to `develop` (using the v2 compose file).

The compose template is no longer used, so the `make init` step is no longer required and has been removed 

In addition, docker-related commands and filenames have been modernized, and some orphan files and references have been removed:

- replaced docker compose template by separate dockerfiles for fdp v1 and v2
  - expose ports to localhost only
  - added healthchecks
  - removed restart
  - changed fdp port from 3000 to spring default 8080
- replaced `application.yml` file by env variables in compose files
- replaced `*_IMAGE` env variables by `*_VERSION`, for convenience
- added `npm run test` for convenience
- updated `Makefile` to use modern docker compose calls and select compose file based on version
- updated `README` to reflect the changes
- removed orphaned OpenRefine references (OpenRefine tests were removed in 8e57d40)

fixes #13

[1]: https://github.com/FAIRDataTeam/FAIRDataPoint/blob/d0ca93f0ab993450a53f47bd3fd57270180c24f6/src/main/resources/application.yml#L15
[2]: https://github.com/FAIRDataTeam/FAIRDataPoint/blob/34fb69794caddb7dccb0137e047303bdef50bb2c/src/main/resources/application.yml#L11

MarekSuchanek commented on Makefile:
```diff
@@ -45,9 +45,9 @@ ci:
 .PHONY: clean
 clean:
-	@rm -rf output && rm -rf fdp/graphdb && rm -f fdp/docker-compose.yml && rm -rf fdp/openrefine-cfg/*.yaml
+	@rm -rf output && rm -rf fdp/graphdb && rm -f fdp/compose.yml && rm -rf fdp/openrefine-cfg/*.yaml
```
openrefine can be deleted here as well?

dennisvang replied:
Done. I also removed other remaining references to openrefine.

MarekSuchanek commented on README.md:
```diff
@@ -9,7 +9,7 @@
 - `/fdp`
     - Contains the configuration to run FDP and other associated services in Docker.
-    - The actual docker-compose.yml file is generated during initialization.
+    - The actual docker compose.yml file is generated during initialization.
```
Docker `compose.yml` file?

dennisvang replied:
Done.

MarekSuchanek commented on README.md:
```diff
@@ -19,27 +19,36 @@
 
 - `make install` - install all the dependencies to run the tests
-- `make init` - initialize docker-compose so that the project can run
+- `make init` - initialize docker compose file so that the project can run
```
same as above?

dennisvang replied:
Done.

dennisvang commented:
@MarekSuchanek Thanks for the review. :)

I ran into some additional issues, so I did some more cleanup and replaced the compose template by separate compose files. As a result the `make init` step is no longer required, so I removed that.

The end result is the same as before. 

Do you want to review the changes again, or can I just merge it?

dennisvang commented:
Thanks again @MarekSuchanek :)
